server/src/com/thoughtworks/go/server/materials/DependencyMaterialUpdateNotifier.java

```diff
@@ -121,26 +140,30 @@ private void updateMaterialsOnIntialization() {
         }
     }
 
-    private void scheduleNewMaterialsForUpdate() {
+    private void scheduleRecentlyAddedMaterialsForUpdate() {
+        for (Object material : newDependencyMaterials()) {
+            updateMaterial((Material) material);
+        }
+    }
+
+    private Collection newDependencyMaterials() {
```

Comment:
The fact that this has a side-effect of changing the `this.dependencyMaterials` is not obvious from this name.